### DISCUSSION TITLE
Taller Pandas - Task 3: Fill in missing data

### DISCUSSION THREAD

**Discussion by Rh0dwulf** [rramosp/ai4eng.v1#15] [02 - PYTHON]
Buenos días profesor Raúl. Cordial saludo y gracias por el curso. 

Estaba desarrollando el taller de pandas y me he encontrado con un inconveniente con el punto 3.

Esta es la función que construí para solucionar el ejercicio: 

![image](https://user-images.githubusercontent.com/6087406/119841564-8ab9c100-becb-11eb-8c65-72948b7eaf7f.png)

Al probar el código manualmente este funciona y no presenta ningún problema: 

![image](https://user-images.githubusercontent.com/6087406/119841772-b8066f00-becb-11eb-82e0-008d000b69bf.png)
![image](https://user-images.githubusercontent.com/6087406/119841807-c05eaa00-becb-11eb-9766-e2c9be062f7e.png)

Sin embargo, al enviar la tarea a la plataforma se genera un error al ejecutar el código, y no entiendo realmente por qué. 
El error que suelta al final es: "ValueError: cannot reindex from a duplicate axis"

![image](https://user-images.githubusercontent.com/6087406/119842159-0e73ad80-becc-11eb-9430-ef8bae5e4fbc.png)

**Comment by rramosp** [comment]
hola rodolfo, te sugiero que no hagas un fillna, si no que explícitamente selecciones los valores na que quieres modificar.

`df['price'][df.price.isna()] = newvals`

ten en cuenta que el corrector genera muchos casos aleatorios con los que comprueba tu código y puede que se dé alguna condición extraña en los índices.

pero es cierto que el mensaje es un poco críptico. trataremos de mejorar el mensaje del corrector.

espero que con esto te funcione. cuéntamos si es así

**Reply by rramosp** [reply]
de hecho, no puedes asumir que el index sea único ... creo que tu código falla cuando hay varias filas en el dataframe con el mismo index y con nan

**Reply by rsbgonzalez** [reply]
Profe, intenté utilizando df['price'][df.price.isna()] = newvals en mi código, pero no me cambia los valores Nan.

pm = d.mean()['price']
pstd = d.std()['price']
Nan = len(d) - d.count()[0]
indice = np.arange(1,len(d)+1)
sample = np.random.normal(pm,pstd,len(d))
samples = pd.Series(sample)
d['price'][d.price.isna()] = samples
result = d
